discovery/aws/ec2_test.go

```diff
@@ -0,0 +1,467 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+// "Helper" functions.
+const (
+	chars     = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	ipv6chars = "0123456789abcdef"
+)
+
+func RandIpv4() string {
```

Comment:
all these utility  test functions (`RandIpv4`, `RandIpv6`....)  don't need to be public, and hence they don't need to start with a capital letter. Neither does the `EC2Data` type.

Reply:
(Hopefully) fixed in https://github.com/prometheus/prometheus/pull/14364/commits/b72467f305c9286142d099c37eef4214b3b41ee8.

What about the `GenerateTest*` functions. Should I make those private as well?